lib/src/kvs/clock.rs

```diff
@@ -0,0 +1,97 @@
+use crate::dbs::node::Timestamp;
+use crate::sql;
+use sql::Duration;
+use std::sync::{Arc, RwLock};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+// Traits cannot have async and we need sized structs for Clone + Send + Sync
+#[derive(Clone)]
+pub enum SizedClock {
+	Fake(FakeClock),
+	Inc(IncFakeClock),
+	System(SystemClock),
+}
+
+impl SizedClock {
+	pub async fn now(&mut self) -> Timestamp {
+		match self {
+			SizedClock::Fake(c) => c.now(),
+			SizedClock::Inc(c) => c.now().await,
+			SizedClock::System(c) => c.now(),
+		}
+	}
+}
+
+/// FakeClock is a clock that is fully controlled externally.
+/// Use this clock for when you are testing timestamps.
+#[derive(Clone)]
+pub struct FakeClock {
+	now: Arc<RwLock<Timestamp>>,
+}
+
+impl FakeClock {
+	pub fn new(now: Timestamp) -> Self {
+		FakeClock {
+			now: Arc::new(RwLock::new(now)),
+		}
+	}
+
+	pub fn now(&self) -> Timestamp {
+		*self.now.read().unwrap()
+	}
+
+	pub fn set(&mut self, timestamp: Timestamp) {
+		self.now.write().unwrap().set(timestamp);
+	}
+}
+
+/// IncFakeClock increments a local clock every time the clock is accessed, similar to a real clock.
+/// This is useful when you need unique and partially deterministic timestamps for tests.
+/// Partially deterministic, because you do not have direct control over how many times a clock
+/// is accessed, and due to the nature of async - you neither have order guarantee.
+#[derive(Clone, Copy)]
+pub struct IncFakeClock {
+	now: Timestamp,
+	increment: Duration,
+}
+
+impl IncFakeClock {
+	pub fn new(now: Timestamp, increment: Duration) -> Self {
+		IncFakeClock {
+			now,
+			increment,
+		}
+	}
+
+	pub async fn now(&mut self) -> Timestamp {
+		self.now = &self.now + &self.increment;
+		self.now
+	}
+}
+
+/// SystemClock is a clock that uses the system time.
+/// Use this when there are no other alternatives.
+#[derive(Clone)]
+pub struct SystemClock {}
```

Comment:
Do we plan to add fields to this struct? If not, we can just do

```rust
pub struct SystemClock;
```

instead, which can be instantiated using just `SystemClock` instead of `SystemClock {}`.